scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -489,7 +505,7 @@ void BPF_STRUCT_OPS(rustland_update_idle, s32 cpu, bool idle)
 	 * Moreover, kick the CPU to make it immediately ready to accept
 	 * dispatched tasks.
 	 */
-	if (__sync_fetch_and_add(&nr_queued, 0)) {
+	if (nr_queued || nr_scheduled) {
```

Comment:
It may be worth explaining how this is interlocked with userland component and a situation where userland component is not invoked even while there are tasks to process can't happen. nr_queued is basically used as a boolean here indicating "something happened since userland looked at it, so better call them at least once". And then while the userland is running it updates its internal state through nr_scheduled. As long as update_idle is called after each userland run, this should be safe, right?

Reply:
@htejun comment updated adding more details. Let me know if it seems clear enough. Thanks!